community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/planner/logical/idp/SingleComponentPlanner.scala

```diff
@@ -92,7 +92,16 @@ case class SingleComponentPlanner(monitor: IDPQueryGraphSolverMonitor,
     for (pattern <- qg.patternRelationships)
       yield {
         val accessPlans = planSinglePattern(qg, pattern, leaves).map(plan => kit.select(plan, qg))
-        val bestAccessor = kit.pickBest(accessPlans).getOrElse(throw new InternalException("Found no access plan for a pattern relationship in a connected component. This must not happen."))
+        val relationships = qg.patternRelationships.map(_.name)
+        val argumentContainsRelationship = qg.argumentIds.exists(relationships.contains)
+        val plans = if (argumentContainsRelationship) {
+          accessPlans
+        } else {
+          val maxSize = accessPlans.map(_.availableSymbols.size).max
```

Comment:
Future me - and to some extent present me - will be confused why we do this. Can we have a short comment explaining the rationale behind these lines of code.

Reply:
Thanks for making me think! When I revisited this, it was clear that this was a hack and not the real solution. I take it back!